File: 412_Fizz_Buzz/xsy/solution.py
class Solution:
    def fizzBuzz(self, n):
        """
        :type n: int
        :rtype: List[str]
        """
        return [str(i) if (i%3 and i%5) else "Fizz"*(i%3==0)+"Buzz"*(i%5==0) for i in range(1, n+1)]

File: 412_FIzz_Buzz/xsy/solution.go
package main

import (
	"strconv"
	"fmt"
)

func fizzBuzz(n int) []string {
    s := []string{}
    tree := 0
    five := 0
    for i := 1; i<=n; i++ {
        tree += 1
        five += 1
        var t string
        if tree == 3 {
            t += "Fizz"
            tree = 0
        } 
        if five == 5 {
            t += "Buzz";
            five = 0
        } 
        if t == "" {
            t = strconv.Itoa(i)
        }
        s = append(s, t)
    }
    return s
}


func main() {
    fmt.Println(fizzBuzz(15))
}
